docs/modules/ROOT/pages/quarkus-azure-cosmos.adoc

```diff
@@ -0,0 +1,218 @@
+
+==== Use the CosmosClient in an imperative application
+
+The `createItem` method first creates the item with request payload in the specified database and container of the Azure Cosmos DB account.
+The `getItem` method reads the item with the id from the specified database and container of the Azure Cosmos DB account.
+
+[source,java]
+----
+@Path("/quarkus-azure-cosmos")
+@ApplicationScoped
+public class CosmosResource {
+
+    @Inject
+    CosmosClient cosmosClient;
+
+    @Path("/{database}/{container}")
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response createItem(
+            @PathParam("database") String database,
+            @PathParam("container") String container,
+            Item body,
+            @Context UriInfo uriInfo) {
+
+        getContainer(database, container).upsertItem(body);
+        return Response.created(uriInfo.getAbsolutePathBuilder().path(body.getId()).build()).build();
+    }
+
+    @Path("/{database}/{container}/{itemId}")
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getItem(
+            @PathParam("database") String database,
+            @PathParam("container") String container,
+            @PathParam("itemId") String itemId) {
+        CosmosItemResponse<Item> item = getContainer(database, container).readItem(itemId, new PartitionKey(itemId),
+                Item.class);
+
+        return Response.ok().entity(item.getItem()).build();
+    }
+}
+----
+
+To test this sample you can run the following cURL commands after the application is started:
+
+* `curl http://localhost:8080/quarkus-azure-cosmos/demodb/democontainer -X POST -d '{"id": "1", "name": "sync-item"}' -H "Content-Type: application/json"`
+* `curl http://localhost:8080/quarkus-azure-cosmos/demodb/democontainer/1 -X GET`
+
+You can go back to the http://portal.azure.com/[Azure portal], open **Data Explorer** of the Azure Cosmos DB account, and see the item that you've created.
+
+image::quarkus-azure-cosmos-azure-portal2.png[alt=Azure Portal showing the content of item created with the CosmosClient object]
+
+==== Use the CosmosAsyncClient in a reactive application
+
+Similarly, the `createItem` method first asynchronously creates the item with request payload in the specified database and container of the Azure Cosmos DB account.
+The `getItem` method asynchronously reads the item with the id from the specified database and container of the Azure Cosmos DB account.
+
+[source,java]
+----
+@Path("/quarkus-azure-cosmos-async")
+@ApplicationScoped
+public class CosmosAsyncResource {
+
+    @Inject
+    CosmosAsyncClient cosmosAsyncClient;
+
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Uni<Response> createItem(
+            @PathParam("database") String database,
+            @PathParam("container") String container,
+            Item body,
+            @Context UriInfo uriInfo) {
+
+        Mono<CosmosItemResponse<Item>> response = getContainer(database, container)
```

Comment:
Same question for `getContainer`.

Reply:
Fixed in a249fe4d89b84137f7379b6eea30708b8a9a4d1b as well.